src/App.tsx

```diff
@@ -1,26 +1,257 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import * as service from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/TodoItem/TodoItem';
 
-const USER_ID = 0;
+enum Filter {
+  ALL = 'ALL',
+  ACTIVE = 'ACTIVE',
+  COMPLETED = 'COMPLETED',
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterTodos, setFilterTodos] = useState<Filter>(Filter.ALL);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [todoTitle, setTodoTitle] = useState<string>('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+
+  useEffect(() => {
+    service
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'));
+  }, []);
+
+  useEffect(() => {
+    if (errorMessage) {
+      const timer = setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+
+      return () => clearTimeout(timer);
+    }
+
+    return undefined;
+  }, [errorMessage]);
+
+  const filteredTodos = todos.filter(todo => {
+    if (filterTodos === 'ACTIVE') {
+      return !todo.completed;
+    } else if (filterTodos === 'COMPLETED') {
+      return todo.completed;
+    }
+
+    return true;
+  });
+
+  const todosAmount = todos.filter(todo => !todo.completed).length;
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  if (!service.USER_ID) {
     return <UserWarning />;
   }
 
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setErrorMessage('');
+    setTodoTitle(event.target.value);
+  };
+
+  const addTodo = ({ title }: { title: string }) => {
+    if (title.trim() === '') {
```

Comment:
```suggestion
    if (!title.trim()) {
```
empty string is a falsy value